python/ray/train/base_trainer.py

```diff
@@ -37,6 +35,14 @@
 from ray.util.annotations import DeveloperAPI
 from ray._private.dict import merge_dicts
 
+
+from ray.train._internal.storage import _use_storage_context
+
+if _use_storage_context():
+    from ray.train._checkpoint import Checkpoint
+else:
+    from ray.air import Checkpoint
```

Comment:
ok to replace with `ray.train.Checkpoint` fully. Since the isinstance check will be correctly gated by FF.

Reply:
Oh I don't think this worked for me because of some kind of import ordering - I can try again

Reply:
Okay I tried moving the definition higher in `train/__init__.py` - it worked for me locally after doing this, but let me see how this plays out in tests.